src/connection/HttpConnection.ts

```diff
@@ -220,30 +225,36 @@ export default class HttpConnection extends BaseConnection {
 
         this.diagnostic.emit('deserialization', null, options)
         response.on('data', onData)
-        response.on('error', onEnd)
         response.on('end', onEnd)
+        response.on('close', onResponseClose)
```

Comment:
I can see the `onResponseClose()` is throwing `Response aborted while reading the body`, which is also emitted on an `EPIPE` (shown below in line 255).  These two messages should be distinct, since we won't know which one caused this message.

It also appears we don't remove this listener.  It seems like we should remove it in the same places we remove `data` and `end`, most specifically on `onEnd()`.  I'm wondering if the listener is going off when the socket closes, making it seem like an otherwise successful call failed.

Reply:
They're both `EPIPE` errors, which is why I gave them the same error message. It's the same symptom, just happening under slightly different circumstances: one gets a response body back, the other doesn't. That difference requires you to inspect the error's `meta` property to tell the difference, but I can use a slightly different message to help distinguish them further.

While the 'close' event should be implicitly cleaned up during garbage collection, it won't hurt to explicitly remove those listeners at the end of the lifecycle.

I've made both of these changes in https://github.com/elastic/elastic-transport-js/pull/266. Please take a look!